core/src/main/scala/org/apache/spark/network/netty/NettyBlockTransferService.scala

```diff
@@ -139,14 +139,7 @@ private[spark] class NettyBlockTransferService(
           }
         }
       }
-
-      if (maxRetries > 0) {
-        // Note this Fetcher will correctly handle maxRetries == 0; we avoid it just in case there's
-        // a bug in this code. We should remove the if statement once we're sure of the stability.
```

Comment:
I wonder how do you prove this since `maxRetries` default to 3 rather than 0.

Reply:
There are several place overwrite the conf to `0`, please search `("spark.shuffle.io.maxRetries", "0")` in `ExternalShuffleIntegrationSuite` and `BlockManagerSuite`

Reply:
When `spark.shuffle.io.maxRetries=0`, it tests `OneForOneBlockFetcher` rather than `RetryingBlockTransferor`, right?

Reply:
Sound like add unit test cases in `RetryingBlockTransferorSuite` is more reasonable, will update soon

Reply:
Updated UT, please take a look again.